client/src/components/pages/MainChannel.tsx

```diff
@@ -135,58 +235,90 @@ const ChannelsFragment: FC<ChannelProps> = ({channel, searchArgs}) => {
   }, [data?.channels?.edges]);
 
   return (
-    <FlatList
-      scrollIndicatorInsets={{right: 1}}
-      style={{
-        alignSelf: 'stretch',
-      }}
-      contentContainerStyle={
-        channels.length === 0
-          ? {
-              flex: 1,
-              alignSelf: 'stretch',
-              alignItems: 'center',
-              justifyContent: 'center',
-            }
-          : null
-      }
-      keyExtractor={(_, index): string => index.toString()}
-      // @ts-ignore
-      data={channels}
-      renderItem={renderItem}
-      ListHeaderComponent={
-        !bannerError && orientation === Orientation.PORTRAIT
-          ? Platform.select({
-              android: (
-                <AdMobBanner
-                  bannerSize={'smartBannerPortrait'}
-                  // adUnitID="ca-app-pub-3940256099942544/6300978111"
-                  adUnitID="ca-app-pub-7837089095803162/8109702961"
-                  onDidFailToReceiveAdWithError={() => setBannerError(true)}
-                />
-              ),
-              ios: (
-                <AdMobBanner
-                  bannerSize={'smartBannerPortrait'}
-                  // adUnitID="ca-app-pub-3940256099942544/2934735716"
-                  adUnitID="ca-app-pub-7837089095803162/4326063134"
-                  onDidFailToReceiveAdWithError={() => setBannerError(true)}
-                />
-              ),
-            })
-          : null
-      }
-      ListEmptyComponent={
-        <EmptyListItem>{getString('NO_CHANNELLIST')}</EmptyListItem>
-      }
-      ListFooterComponent={<View style={{height: 60}} />}
-      refreshing={isLoadingNext}
-      onRefresh={() => {
-        refetch(searchArgs, {fetchPolicy: 'network-only'});
-      }}
-      onEndReachedThreshold={0.1}
-      onEndReached={onEndReached}
-    />
+    <View>
+      <Modal animationType="slide" transparent={true} visible={modalVisible}>
+        <ModalContainer>
+          <ModalViewContainer>
+            <ModalText>{getString('LEAVE_CHANNEL')}</ModalText>
```

Comment:
#### Should Fix
I think this modal should be extracted as a separate component. Please look at `ProfileModal` to see how it is implemented.

Reply:
Thank you for your advice.

As you said, this modal was extracted as a separate component.

I want to bring back the channel list when the user leaves the channel.

How can I get the channel list back?

Reply:
@woosanggyu There are two ways to achieve this:
1. Refetch the channel list.
2. Update the cached value from Relay store. You can look at some examples in `client/src/relay/updaters.tsx` file.

Reply:
i can resolve this problem with your support. thanks ;D